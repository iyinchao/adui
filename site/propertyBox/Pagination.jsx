import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Button, Switch, Pagination } from "../../components"
import { SIZES } from "./consts"

export default class PropertyBox extends React.Component {
  state = {
    showInput: false,
    size: "small",
    theme: false,
  }

  render() {
    const { showInput, size, theme } = this.state

    const codes = `<Pagination
  defaultCurrent={1}
  pageSize={10}
  showButtonJumper
  ${showInput ? `showInputJumper` : ""}
  ${size === "small" ? "" : `size="${size}"`}
  ${theme ? `theme="light"` : ""}
  total={99}
/>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left}>
            <Pagination
              showInputJumper={showInput}
              theme={theme ? "light" : null}
              size={size}
              align="center"
              defaultCurrent={1}
              pageSize={10}
              total={99}
              onChange={(current, pageSize) => console.log(current, pageSize)}
              showButtonJumper
            />
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>尺寸</dt>
              <dd className={styles.controlContent}>
                <Button.Group banner>
                  {SIZES.map(({ value, text }) => (
                    <Button
                      key={value}
                      active={value === size}
                      onClick={() => this.setState({ size: value })}
                    >
                      {text}
                    </Button>
                  ))}
                </Button.Group>
              </dd>
            </dl>
            <dl className={styles.controls}>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="可快捷键跳转"
                  unCheckedText="可快捷键跳转"
                  checked={showInput}
                  onChange={(param) => this.setState({ showInput: param })}
                />
              </div>
              <div className={styles.switchControl}>
                <Switch
                  checkedText="轻量风格"
                  unCheckedText="轻量风格"
                  checked={theme}
                  onChange={(param) => this.setState({ theme: param })}
                />
              </div>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
