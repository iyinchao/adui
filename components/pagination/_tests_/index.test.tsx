/* eslint-disable */
import React, { useState } from "react"
import { render, mount } from "enzyme"
import TestRenderer from "react-test-renderer"
import sinon from "sinon"
import Pagination from ".."

const PaginationMounter = () => {
  const [current, setCurrent] = useState(1)
  return (
    <Pagination
      pageSize={10}
      current={current}
      total={99}
      onChange={(newCurrent) => setCurrent(newCurrent)}
      showButtonJumper
    />
  )
}

describe("Pagination", () => {
  it("是否正确渲染", () => {
    const wrapper = TestRenderer.create(
      <Pagination
        align="right"
        defaultCurrent={1}
        pageSize={10}
        total={99}
        showButtonJumper
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("current Prop 是否能验证非整数", () => {
    const stub = sinon.stub(console, "error")
    render(<Pagination current={3.1} />)
    expect(stub.calledTwice).toEqual(true)
    stub.restore()
  })

  it("内部驱动", () => {
    const wrapper = mount(
      <Pagination
        align="right"
        defaultCurrent={5}
        pageSize={10}
        total={99}
        showButtonJumper
        showInputJumper
      />
    )
    expect(wrapper.find("div").at(0).props()["data-current"]).toEqual(5)
    wrapper.find("button").at(1).simulate("click")
    expect(wrapper.find("div").at(0).props()["data-current"]).toEqual(4)

    wrapper.find("button").at(0).simulate("click")
    expect(wrapper.find("div").at(0).props()["data-current"]).toEqual(1)

    const input = wrapper.find("input")
    input.simulate("change", { target: { value: "3" } })
    input.simulate("keydown", { keyCode: 13 })
    expect(wrapper.find("div").at(0).props()["data-current"]).toEqual(3)

    input.simulate("change", { target: { value: "你好" } })
    input.simulate("keydown", { keyCode: 13 })
  })

  it("外部驱动", () => {
    const wrapper1 = mount(
      <Pagination
        align="right"
        pageSize={10}
        current={1}
        total={99}
        showButtonJumper
      />
    )
    wrapper1.find("button").at(2).simulate("click")
    expect(wrapper1.find("div").at(0).props()["data-current"]).toEqual(1)

    wrapper1.find("button").at(3).simulate("click")
    expect(wrapper1.find("div").at(0).props()["data-current"]).toEqual(1)
  })

  const wrapper2 = mount(<PaginationMounter />)
  wrapper2.find("button").at(2).simulate("click")
  expect(wrapper2.find("div").at(0).props()["data-current"]).toEqual(2)
})
