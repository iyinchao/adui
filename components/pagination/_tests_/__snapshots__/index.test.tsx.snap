// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 是否正确渲染 1`] = `
<div
  className="adui-pagination-base adui-pagination-small adui-pagination-right"
  data-current={1}
>
  <button
    className="adui-button-base adui-button-normal adui-button-small adui-button-hasLeftIcon adui-button-hasRightIcon"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "8px",
      }
    }
    type="button"
  >
    <svg
      className="adui-button-leftIcon adui-icon-base"
      data-icon="arrow-left-first"
      data-interactive={false}
      fill="var(--gray-700)"
      height={18}
      onClick={[Function]}
      style={Object {}}
      viewBox="0 0 18 18"
      width={18}
    >
      <path
        d="M4.99999 3.5C4.72385 3.5 4.49999 3.72386 4.49999 4V14C4.49999 14.2761 4.72385 14.5 4.99999 14.5H5.49999C5.77613 14.5 5.99999 14.2761 5.99999 14V4C5.99999 3.72386 5.77613 3.5 5.49999 3.5H4.99999ZM13.3 3.69645C13.1264 3.52288 12.857 3.50359 12.6621 3.63859L12.5929 3.69645L7.64313 8.64619C7.46957 8.81976 7.45028 9.08918 7.58528 9.28405L7.64313 9.3533L12.5929 14.303C12.7664 14.4766 13.0359 14.4959 13.2307 14.3609L13.3 14.303L13.6535 13.9495C13.8271 13.7759 13.8464 13.5065 13.7114 13.3116L13.6535 13.2424L9.41069 8.999L13.6535 4.75711C13.8271 4.58354 13.8464 4.31412 13.7114 4.11925L13.6535 4.05L13.3 3.69645Z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <button
    className="adui-button-base adui-button-normal adui-button-small adui-button-hasLeftIcon adui-button-hasRightIcon"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="adui-button-leftIcon adui-icon-base"
      data-icon="arrow-left"
      data-interactive={false}
      fill="var(--gray-700)"
      height={18}
      onClick={[Function]}
      style={Object {}}
      viewBox="0 0 18 18"
      width={18}
    >
      <path
        d="M10.8033 3.69643L11.1569 4.04999C11.3521 4.24525 11.3521 4.56183 11.1569 4.75709L6.914 8.99899L11.1569 13.2424C11.3521 13.4376 11.3521 13.7542 11.1569 13.9495L10.8033 14.303C10.608 14.4983 10.2915 14.4983 10.0962 14.303L5.14645 9.35329C4.95118 9.15803 4.95118 8.84144 5.14645 8.64618L10.0962 3.69643C10.2915 3.50117 10.608 3.50117 10.8033 3.69643Z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <span
    className="adui-pagination-text"
  >
    1
     / 
    10
  </span>
  <button
    className="adui-button-base adui-button-normal adui-button-small adui-button-hasLeftIcon adui-button-hasRightIcon"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="adui-button-leftIcon adui-icon-base"
      data-icon="arrow-right"
      data-interactive={false}
      fill="var(--gray-700)"
      height={18}
      onClick={[Function]}
      style={Object {}}
      viewBox="0 0 18 18"
      width={18}
    >
      <path
        d="M7.20001 3.69645L6.84646 4.05C6.6512 4.24527 6.6512 4.56185 6.84646 4.75711L11.0893 8.999L6.84646 13.2424C6.6512 13.4377 6.6512 13.7542 6.84646 13.9495L7.20001 14.3031C7.39527 14.4983 7.71186 14.4983 7.90712 14.3031L12.8569 9.3533C13.0521 9.15804 13.0521 8.84146 12.8569 8.6462L7.90712 3.69645C7.71186 3.50119 7.39527 3.50119 7.20001 3.69645Z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <button
    className="adui-button-base adui-button-normal adui-button-small adui-button-hasLeftIcon adui-button-hasRightIcon"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginLeft": "8px",
      }
    }
    type="button"
  >
    <svg
      className="adui-button-leftIcon adui-icon-base"
      data-icon="arrow-right-last"
      data-interactive={false}
      fill="var(--gray-700)"
      height={18}
      onClick={[Function]}
      style={Object {}}
      viewBox="0 0 18 18"
      width={18}
    >
      <path
        d="M13.3 3.5C13.5761 3.5 13.8 3.72386 13.8 4V14C13.8 14.2761 13.5761 14.5 13.3 14.5H12.8C12.5239 14.5 12.3 14.2761 12.3 14V4C12.3 3.72386 12.5239 3.5 12.8 3.5H13.3ZM5 3.69645C5.17357 3.52288 5.44299 3.50359 5.63786 3.63859L5.70711 3.69645L10.6569 8.64619C10.8304 8.81976 10.8497 9.08918 10.7147 9.28405L10.6569 9.3533L5.70711 14.303C5.53354 14.4766 5.26412 14.4959 5.06925 14.3609L5 14.303L4.64645 13.9495C4.47288 13.7759 4.4536 13.5065 4.58859 13.3116L4.64645 13.2424L8.8893 8.999L4.64645 4.75711C4.47288 4.58354 4.4536 4.31412 4.58859 4.11925L4.64645 4.05L5 3.69645Z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
